### ISSUE TITLE
Some questions about LLM training

### ISSUE THREAD

**Issue by bo-wang-up** [MIV-XJTU/JanusVLN#37]
Hi,
I'm finetuning JanusVLN based on your provided weights (JanusVLN_extra)

After including the LLM in training, the loss eventually becomes 0.0.

With gradient checkpointing enabled, this happens almost immediately. With it disabled, the loss still drops to 0.0 after around 30 steps.

After debugging, I found that the gate weights in layer 22 become NaN during training, which seems to cause the issue.

Have you encountered anything similar before?

Thanks!

**Comment by anhdao69** [comment]
I got same issues when training from scratch.

**Comment by phamquandung** [comment]
Did you guys figure out the issue?

**Comment by phamquandung** [comment]
> Hi, I'm finetuning JanusVLN based on your provided weights (JanusVLN_extra)
> 
> After including the LLM in training, the loss eventually becomes 0.0.
> 
> With gradient checkpointing enabled, this happens almost immediately. With it disabled, the loss still drops to 0.0 after around 30 steps.
> 
> After debugging, I found that the gate weights in layer 22 become NaN during training, which seems to cause the issue.
> 
> Have you encountered anything similar before?
> 
> Thanks!

The problem is all about the configration of the zero2.json file.

**Comment by bo-wang-up** [comment]
> > Hi, I'm finetuning JanusVLN based on your provided weights (JanusVLN_extra)
> > After including the LLM in training, the loss eventually becomes 0.0.
> > With gradient checkpointing enabled, this happens almost immediately. With it disabled, the loss still drops to 0.0 after around 30 steps.
> > After debugging, I found that the gate weights in layer 22 become NaN during training, which seems to cause the issue.
> > Have you encountered anything similar before?
> > Thanks!
> 
> The problem is all about the configration of the zero2.json file.

Thanks for the hint!
Could you please share how you fixed the zero2.json configuration?

I also tried zero3.json, but it seems to run into the same issue as mentioned in #22.
Do you have any recommended DeepSpeed config or training setting for avoiding the NaN / loss=0.0 problem?

Thanks a lot!

**Comment by phamquandung** [comment]
Please use this config for zero2.json. Let me know if it does not work 

```
{
    "fp16": {
        "enabled": false
    },
    "bf16": {
        "enabled": true
    },
    "zero_allow_untested_optimizer": true,
    "data_types": {
        "grad_accum_dtype": "fp32"
    },
    "gradient_clipping": 1.0,
    "train_micro_batch_size_per_gpu": "auto",
    "train_batch_size": "auto",
    "gradient_accumulation_steps": "auto",
    "zero_optimization": {
        "stage": 2,
        "overlap_comm": false,
        "contiguous_gradients": true,
        "sub_group_size": 1e9,
        "reduce_bucket_size": "auto"
    }
}
```

**Comment by bo-wang-up** [comment]
> Please use this config for zero2.json. Let me know if it does not work
> 
> ```
> {
>     "fp16": {
>         "enabled": false
>     },
>     "bf16": {
>         "enabled": true
>     },
>     "zero_allow_untested_optimizer": true,
>     "data_types": {
>         "grad_accum_dtype": "fp32"
>     },
>     "gradient_clipping": 1.0,
>     "train_micro_batch_size_per_gpu": "auto",
>     "train_batch_size": "auto",
>     "gradient_accumulation_steps": "auto",
>     "zero_optimization": {
>         "stage": 2,
>         "overlap_comm": false,
>         "contiguous_gradients": true,
>         "sub_group_size": 1e9,
>         "reduce_bucket_size": "auto"
>     }
> }
> ```

Thanks! This works well

**Comment by RichardChangCA** [comment]
> > Please use this config for zero2.json. Let me know if it does not work
> > ```
> > {
> >     "fp16": {
> >         "enabled": false
> >     },
> >     "bf16": {
> >         "enabled": true
> >     },
> >     "zero_allow_untested_optimizer": true,
> >     "data_types": {
> >         "grad_accum_dtype": "fp32"
> >     },
> >     "gradient_clipping": 1.0,
> >     "train_micro_batch_size_per_gpu": "auto",
> >     "train_batch_size": "auto",
> >     "gradient_accumulation_steps": "auto",
> >     "zero_optimization": {
> >         "stage": 2,
> >         "overlap_comm": false,
> >         "contiguous_gradients": true,
> >         "sub_group_size": 1e9,
> >         "reduce_bucket_size": "auto"
> >     }
> > }
> > ```
> 
> Thanks! This works well

Hi, I would like to know whether the new zero2.json file  works. I use this new zero2.json file, but training is still 0/NAN loss in my side. Thanks in advance!

**Comment by bo-wang-up** [comment]
> > > Please use this config for zero2.json. Let me know if it does not work
> > > ```
> > > {
> > >     "fp16": {
> > >         "enabled": false
> > >     },
> > >     "bf16": {
> > >         "enabled": true
> > >     },
> > >     "zero_allow_untested_optimizer": true,
> > >     "data_types": {
> > >         "grad_accum_dtype": "fp32"
> > >     },
> > >     "gradient_clipping": 1.0,
> > >     "train_micro_batch_size_per_gpu": "auto",
> > >     "train_batch_size": "auto",
> > >     "gradient_accumulation_steps": "auto",
> > >     "zero_optimization": {
> > >         "stage": 2,
> > >         "overlap_comm": false,
> > >         "contiguous_gradients": true,
> > >         "sub_group_size": 1e9,
> > >         "reduce_bucket_size": "auto"
> > >     }
> > > }
> > > ```
> > 
> > 
> > Thanks! This works well
> 
> Hi, I would like to know whether the new zero2.json file works. I use this new zero2.json file, but training is still 0/NAN loss in my side. Thanks in advance!

sorry, I just see your reply. I guess two settings are important, 
"grad_accum_dtype:”fp32”" and cancel "bf 16" and "fp16" auto
